### PR TITLE
Move ir_pre_fusion.txt and ir_post_fusion.txt to TORCH_LOGS

### LINKED ISSUE #147002: [Inductor] Add Torch Logs for ir_pre_fusion and ir_post_fusion
### 🚀 The feature, motivation and pitch

When you run with `TORCH_COMPILE_DEBUG=1` we serialize pre fusion and post fusion IR. See here, https://github.com/pytorch/pytorch/blob/7f62616a585f91a6cccb9672c42bc8210044b1bf/torch/_inductor/debug.py#L524-L528. 

TORCH_COMPILE_DEBUG is kind of an earlier mechanism that has mostly been replaced with a combination of TORCH_LOGS and tlparse. We should add a similar TORCH_LOGS="ir_pre_fusion" and TORCH_LOGS="ir_post_fusion" to more accessibly debug states of the IR. 

Check out the registration of logging [here](https://github.com/pytorch/pytorch/blob/main/torch/_logging/_registrations.py). If you click around blame there should be a PR that shows how to add a logging artifact. 

### Alternatives

_No response_

### Additional context

_No response_

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @aakhundov

### PR DESCRIPTION
**Pull request by dulinriley**
Fixes #147002

Moves ir_{pre, post}_fusion.txt to be controlled by TORCH_LOGS instead of TORCH_COMPILE_DEBUG.
Updated tests of these logs as well.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

### CODE REVIEW

File: test/inductor/test_debug_trace.py [hunk 1]
```diff
@@ -51,20 +56,27 @@ def fn(a, b):
             with self.assertLogs(
                 logging.getLogger("torch._inductor.debug"), level=logging.WARNING
             ) as cm:
-                fn(torch.randn(16, 16), torch.randn(16, 16))
+                with ctx():
+                    fn(torch.randn(16, 16), torch.randn(16, 16))
```
Comment (Skylion007) [inline test/inductor/test_debug_trace.py:60]:
```suggestion
            ) as cm, ctx():
                fn(torch.randn(16, 16), torch.randn(16, 16))

Comment (dulinriley) [reply test/inductor/test_debug_trace.py]: Thanks, done in [255b0f1](https://github.com/pytorch/pytorch/commit/255b0f1afe48e27358d3fb39f20961afdc5ab0dd)

**Comment by dulinriley** [comment]
The `len(cm.output)` check against 1 seems to be failing on CI, but passing when I run the test locally with `python test/run_test.py -i inductor/test_debug_trace`.

Does anyone have an idea why that might be the case? I see CI uses python 3.9 and locally it is using 3.12, but I wouldn't expect that to affect log levels.

File: torch/testing/_internal/logging_utils.py [hunk 2]
```diff
@@ -211,3 +211,33 @@ def ctx_manager():
         return exit_stack
 
     return log_stream, ctx_manager
+
+
+def multiple_logs_to_string(module, *log_options):
```
Comment (eellison) [inline torch/testing/_internal/logging_utils.py:216]: i know this is test code so matters a bit less, but nonetheless: `module: str, *log_options: str) -> Tuple[List[io.StringIO], typing.ContextManager[None]]`

Comment (eellison) [inline torch/testing/_internal/logging_utils.py:216]: helpful utility! i could have used this previously.

**Review by eellison — APPROVED** [review]
nice looks good ! just a couple of small nits

**Comment by dulinriley** [comment]
Applied nits, going to merge

**Comment by FindHao** [comment]
Hi @dulinriley , will this PR totally remove the ir_pre/post_fusion.txt generations? I tried "TORCH_LOGS": "+ir_pre_fusion,ir_post_fusion" and it doesn't generate those txt files. am I using the wrong flags?

**Comment by dulinriley** [comment]
I just checked, you still need to add TORCH_COMPILE_DEBUG=1 to your environment variables, else the `DebugContext` object doesn't exist and doesn't call the ir_pre_fusion function.

I verified with https://github.com/pytorch/pytorch/pull/147560:
```
TORCH_LOGS="ir_pre_fusion" TORCH_COMPILE_DEBUG=1 python ./.ci/pytorch/smoke_test/smoke_test.py
```

(I added a torch.compile call in that smoke test to get a quick example. And that is with info logs on).
Since that whole file is based around TORCH_COMPILE_DEBUG, I should've mentioned in this PR that it wasn't getting rid of that

**Comment by FindHao** [comment]
> I just checked, you still need to add TORCH_COMPILE_DEBUG=1 to your environment variables, else the `DebugContext` object doesn't exist and doesn't call the ir_pre_fusion function.
> 
> I verified with #147560:
> 
> ```
> TORCH_LOGS="ir_pre_fusion" TORCH_COMPILE_DEBUG=1 python ./.ci/pytorch/smoke_test/smoke_test.py
> ```
> 
> (I added a torch.compile call in that smoke test to get a quick example. And that is with info logs on). Since that whole file is based around TORCH_COMPILE_DEBUG, I should've mentioned in this PR that it wasn't getting rid of that

I still can't see the ir_pre_fusion.txt. can you try to following code and commands?

```python
import torch

# Convert the function to use PyTorch tensors
def fn_torch(x):
    return x * x

# Compile the function using torch.compile
fn_compiled = torch.compile(fn_torch)

# Example usage
x = torch.tensor([2.0, 3.0, 4.0])
result = fn_compiled(x)
print(result)
```
command to run `TORCH_LOGS="ir_pre_fusion,ir_post_fusion" TORCH_COMPILE_DEBUG=1 python a.py`.

**Comment by dulinriley** [comment]
I ran your sample on this commit: a821d69d925de1bcde05439b497867397bbcd7b0
And I saw this output:
```
$ TORCH_LOGS="ir_pre_fusion,ir_post_fusion" TORCH_COMPILE_DEBUG=1 python ./a.py
W0225 13:23:49.337000 2391123 torch/_inductor/debug.py:454] [0/0] model__0_inference_0 debug trace: /pytorch/torch_compile_debug/run_2025_02_25_13_23_44_082087-pid_2391123/torchinductor/model__0_inference_0.0

I0225 13:24:06.617000 2394794 torch/_inductor/debug.py:531] [0/0] [__ir_pre_fusion] BEFORE FUSION
I0225 13:24:06.617000 2394794 torch/_inductor/debug.py:531] [0/0] [__ir_pre_fusion] op0: SchedulerNode(ComputedBuffer)
I0225 13:24:06.617000 2394794 torch/_inductor/debug.py:531] [0/0] [__ir_pre_fusion] op0.writes = [MemoryDep('buf0', c0, {c0: 3})]
... many more lines
I0225 13:24:06.624000 2394794 torch/_inductor/debug.py:535] [0/0] [__ir_post_fusion] AFTER FUSION
I0225 13:24:06.624000 2394794 torch/_inductor/debug.py:535] [0/0] [__ir_post_fusion] op0: SchedulerNode(ComputedBuffer)
I0225 13:24:06.624000 2394794 torch/_inductor/debug.py:535] [0/0] [__ir_post_fusion] op0.writes = [MemoryDep('buf0', c0, {c0: 3})]
```

I don't see these get written to any file, but I assume that TORCH_LOGS for artifact loggers might have some redirection API you could use:
https://pytorch.org/docs/main/logging.html
Or you can redirect stderr.

One thing to note: for some reason I don't get this output unless I add any text to `torch/_inductor/debug.py`. This includes even just a comment in a different function or at the top of the file. It also works if I add a comment to `torch/_inductor/scheduler.py`.
So I assume this has something to do with the python build setup maybe using cached Python bytecode,
and would not be reflected in a release binary. I can try and download an alpha build wheel and see if the same effect
occurs.
